release-notes/9.0/9.0.0/api-diff/Microsoft.NETCore.App/9.0.0_System.Diagnostics.Metrics.md

```diff
@@ -0,0 +1,45 @@
+# System.Diagnostics.Metrics
+
+``` diff
+ namespace System.Diagnostics.Metrics {
++    public sealed class Gauge<T> : Instrument<T> where T : struct {
++        public void Record(T value);
++        public void Record(T value, KeyValuePair<string, object?> tag);
++        public void Record(T value, KeyValuePair<string, object?> tag1, KeyValuePair<string, object?> tag2);
++        public void Record(T value, KeyValuePair<string, object?> tag1, KeyValuePair<string, object?> tag2, KeyValuePair<string, object?> tag3);
++        public void Record(T value, params KeyValuePair<string, object>[] tags);
++        public void Record(T value, in TagList tagList);
++        public void Record(T value, ReadOnlySpan<KeyValuePair<string, object?>> tags);
++    }
+     public abstract class Instrument {
++        protected Instrument(Meter meter, string name);
+-        protected Instrument(Meter meter, string name, string? unit, string? description, IEnumerable<KeyValuePair<string, object?>>? tags);
++        protected Instrument(Meter meter, string name, string? unit = null, string? description = null, IEnumerable<KeyValuePair<string, object?>>? tags = null);
+     }
+     public abstract class Instrument<T> : Instrument where T : struct {
++        protected Instrument(Meter meter, string name);
++        protected Instrument(Meter meter, string name, string? unit = null, string? description = null, IEnumerable<KeyValuePair<string, object?>>? tags = null, InstrumentAdvice<T>? advice = null);
++        public InstrumentAdvice<T>? Advice { get; }
+     }
++    public sealed class InstrumentAdvice<T> where T : struct {
++        public InstrumentAdvice();
++        public IReadOnlyList<T>? HistogramBucketBoundaries { get; set; }
++    }
+     public readonly struct Measurement<T> where T : struct {
++        public Measurement(T value, in TagList tags);
+     }
+-    public delegate void MeasurementCallback<T>(Instrument instrument, T measurement, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state);
++    public delegate void MeasurementCallback<T>(Instrument instrument, T measurement, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state) where T : struct;
```

Comment:
I don't think this has changed 

[https://github.com/dotnet/runtime/blob/c1ae962610643a3c17bcf9621017ac42ff7703f7/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterListener.cs#L12](https://github.com/dotnet/runtime/blob/c1ae962610643a3c17bcf9621017ac42ff7703f7/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterListener.cs#L12)